### PR TITLE
Revert ".github: add support for cilium-cli in aws-cni conformance tests"

### PR DESCRIPTION
**Pull request by pchaigno**
Commit https://github.com/cilium/cilium/commit/2d7af3a655e3ed32578acaf9b0d2ced3ade41ec5 (".github: add support for cilium-cli in aws-cni conformance tests") changed the AWS-CNI worflow to install Cilium using the CLI instead of Helm. All of the Helm flags were passed through the CLI using `--helm-set`.

However, the CLI also does it's own magic and isn't aware that we want to install Cilium in chaining mode. Therefore, it detects EKS and incorrectly sets IPAM mode to ENI.

As a result, Cilium attempts to setup the IP rules and routes for new pods. It seems to fail in most cases—maybe because AWS-CNI already setup the state—but sometimes succeeds. When it succeeds, pods end up with an egress rule pointing to a non-existing ip route table.

This usually surfaces in the workflow runs as a DNS failure: the client pod fails to egress to the DNS backend on a remote node. In rarer cases, it surfaces as a failing connectivity test for some of the pods.

This pull request fixes it by reverting https://github.com/cilium/cilium/pull/19555. We unfortunately can't simply set IPAM mode to cluster-pool (default) explicitly because the CLI also uses the wrong operator image (operator-aws instead of operator-generic).

The longer-term fix would be to support chaining mode in the CLI (cf. https://github.com/cilium/cilium-cli/issues/461).

I tested this revert by running the workflow with the test commit ten times at https://github.com/cilium/cilium/runs/6644709056?check_suite_focus=true.

### CODE REVIEW

**Review by aanm — DISMISSED** [review]
> However, the CLI also does it's own magic and isn't aware that we want to install Cilium in chaining mode. Therefore, it detects EKS and incorrectly sets IPAM mode to ENI.

cilium-cli >= v1.11 allows to overwrite any default using the `helm-set` command therefore we could set the right ipam mode that we wish with `--helm-set ipam: kubernetes`

**Comment by pchaigno** [comment]
> > However, the CLI also does it's own magic and isn't aware that we want to install Cilium in chaining mode. Therefore, it detects EKS and incorrectly sets IPAM mode to ENI.
> 
> cilium-cli >= v1.11 allows to overwrite any default using the `helm-set` command therefore we could set the right ipam mode that we wish with `--helm-set ipam: kubernetes`

I tried that. From commit message and PR description:

> We unfortunately can't simply set IPAM mode to cluster-pool (default) explicitly because the CLI also uses the wrong operator image (operator-aws instead of operator-generic).

File: .github/workflows/conformance-aws-cni.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ on:
   # - Don't forget to remove the `DO NOT MERGE` commit once satisfied. The run
   #   will disappear from the PR checks: please provide a direct link to the
   #   successful workflow run (can be found from Actions tab) in a comment.
-  #
+  #
```
Comment (jibi) [inline .github/workflows/conformance-aws-cni.yaml:22]: :eyes:

File: .github/workflows/conformance-aws-cni.yaml [hunk 2]
```diff
@@ -66,7 +66,7 @@ concurrency:
 env:
   clusterName: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
   region: us-east-2
-  cilium_cli_version: v0.11.7
+  cilium_cli_version: v0.10.4
```
Comment (jibi) [inline .github/workflows/conformance-aws-cni.yaml:69]: Why do we need to downgrade the CLI to `0.10`?

File: .github/workflows/conformance-aws-cni.yaml [hunk 3]
```diff
@@ -290,6 +281,13 @@ jobs:
       - name: Post-test information gathering
         if: ${{ !success() }}
         run: |
+          echo "=== Install latest stable CLI ==="
```
Comment (jibi) [inline .github/workflows/conformance-aws-cni.yaml:284]: Why is this needed given that we are already installing the CLI in a previous step (which is also related to my other comment about downgrading it to `0.10`)?

**Review by jibi — APPROVED** [review]
Switching to Helm based installation LGTM, just a couple of questions

File: .github/workflows/conformance-aws-cni.yaml (hunk 1 above)
Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml:22]: Yeah, not sure what happened, but surely not the end of the world. We must have removed this line at some point and the revert added it back.

File: .github/workflows/conformance-aws-cni.yaml (hunk 2 above)
Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml:69]: I'm not sure later versions will work without the `set-helm` flags. I'm sticking to the revert rather than risking breaking other things.

File: .github/workflows/conformance-aws-cni.yaml (hunk 3 above)
Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml:284]: Ah, good point. Missed that part of the revert I guess.
I'd rather improve as a followup PR (which will be needed anyway) than to have to rerun the tests 10 times here, again.

**Comment by aanm** [comment]
> > > However, the CLI also does it's own magic and isn't aware that we want to install Cilium in chaining mode. Therefore, it detects EKS and incorrectly sets IPAM mode to ENI.
> > 
> > 
> > cilium-cli >= v1.11 allows to overwrite any default using the `helm-set` command therefore we could set the right ipam mode that we wish with `--helm-set ipam: kubernetes`
> 
> I tried that. From commit message and PR description:
> 
> > We unfortunately can't simply set IPAM mode to cluster-pool (default) explicitly because the CLI also uses the wrong operator image (operator-aws instead of operator-generic).

yes we can `--operator-image=quay.io/cilium/operator-generic:${SHA}` should set it to the operator generic

**Review by aanm — CHANGES_REQUESTED** [review]
> > > However, the CLI also does it's own magic and isn't aware that we want to install Cilium in chaining mode. Therefore, it detects EKS and incorrectly sets IPAM mode to ENI.
> > 
> > 
> > cilium-cli >= v1.11 allows to overwrite any default using the `helm-set` command therefore we could set the right ipam mode that we wish with `--helm-set ipam: kubernetes`
> 
> I tried that. From commit message and PR description:
> 
> > We unfortunately can't simply set IPAM mode to cluster-pool (default) explicitly because the CLI also uses the wrong operator image (operator-aws instead of operator-generic).

yes we can. `--operator-image=quay.io/cilium/operator-generic:${SHA}` should set it to the operator generic
